.pre-commit-hooks.yaml

```diff
@@ -1,30 +1,61 @@
--   id: puppet-validate
-    name: Validate Puppet manifests
-    description: Validate syntax of Puppet manifests
-    entry: puppet-validate
-    language: ruby
-    files: \.pp$
-    additional_dependencies: ['puppet']
+- id: epp-validate
+  additional_dependencies: ['puppet']
+  description: Validate syntax of Puppet EPP templates
+  entry: epp-validate
+  files: \.epp$
+  language: ruby
+  name: Validate EPP templates
 
--   id: erb-validate
-    name: Validate ERB templates
-    description: Validate syntax of Ruby ERB templates
-    entry: erb-validate
-    language: ruby
-    files: \.erb$
+- id: erb-validate
+  description: Validate syntax of Ruby ERB templates
+  entry: erb-validate
+  files: \.erb$
+  language: ruby
+  name: Validate ERB templates
 
--   id: epp-validate
-    name: Validate EPP templates
-    description: Validate syntax of Puppet EPP templates
-    entry: epp-validate
-    language: ruby
-    files: \.epp$
-    additional_dependencies: ['puppet']
+- id: g10k-validate
+  description: Validate syntax of Puppetfile using g10k
+  entry: g10k-validate
+  files: ^Puppetfile$
+  language: ruby
+  name: Validate g10k Puppetfile
 
--   id: puppet-lint
-    name: puppet-lint
-    description: Check Puppet manifests for stylistic problems
-    entry: puppet-lint
-    language: ruby
-    files: \.pp$
-    additional_dependencies: ['puppet-lint']
+- id: puppet-lint
+  additional_dependencies: ['puppet-lint']
+  description: Check Puppet manifests for stylistic problems
+  entry: puppet-lint
+  files: \.pp$
+  language: ruby
+  name: puppet-lint
+
+- id: puppet-validate
+  additional_dependencies: ['puppet']
+  description: Validate syntax of Puppet manifests
+  entry: puppet-validate
+  files: \.pp$
+  language: ruby
+  name: Validate Puppet manifests
+
+- id: r10k-validate
+  additional_dependencies: ['r10k']
+  description: Validate syntax of Puppetfile using r10k
+  entry: r10k-validate
+  files: ^Puppetfile$
+  language: ruby
+  name: Validate r10k Puppetfile
+
+- id: rubocop
+  additional_dependencies: ['rubocop']
+  description: Check ruby programs for stylistic problems
```

Comment:
Hmm, I wonder if it would make sense to leave out the two generic (non-Puppet-specific) Ruby hooks (`rubocop` and `ruby-validate`) and instead recommend using Ruby hooks out of a separate hook repository, keeping this repo for hooks specific to Puppet.

The main reason I'm thinking this is that the generic Ruby hook repos are likely going to be able to do a better job at staying up-to-date with the best linters, have them configured in the best way, etc. We can avoid duplicating the work they're already doing, and they're likely to provide a better experience for consumers over the long term.

The nice thing about pre-commit is that it's pretty easy to just add one more hook repo (I see several listed on https://pre-commit.com/hooks.html that contain `rubocop`, for example, though I haven't tried any) by just adding another entry in your project's YAML file. We could even suggest a repo in the README here?

Then again... I guess you could make the same argument about `erb-validate` since my understanding is that `erb` isn't really specific to Puppet. Though we probably wouldn't want to remove that one since there doesn't seem to be any existing replacement at the moment.

What do you think?

Reply:
I didn't know it was possible to install multiple hook-repos in the same config-file.

Reply:
Okay; I'll remove the `rubocop` and `ruby-validate` entries, but I think we should reference the preferred versions.

Reply:
By the way, I tried using the [`ruby-lint`](https://github.com/pre-commit/mirrors-ruby-lint) hook, but it chokes on a [custom fact](https://puppet.com/docs/facter/latest/custom_facts.html) and the [upstream project](https://gitlab.com/yorickpeterse/ruby-lint) appears unmaintained.

Reply:
I submitted an issue [here](https://gitlab.com/yorickpeterse/ruby-lint/issues/208).